### PR TITLE
fix: migrate keycard key pairs properly

### LINKED ISSUE #7798: Dropping the keycards table doesn't backfill keypairs.cold_wallet
Upgrading a Keycard profile in place from 2.38.2 to 2.39 loses the record that its accounts are Keycard accounts. The migration that drops the keycards tables doesn't move that fact into `keypairs.cold_wallet`.

## Summary

`1779877215_drop_keycard_tables.up.sql` is two statements:

```sql
DROP TABLE IF EXISTS keycards_accounts;
DROP TABLE IF EXISTS keycards;
```

For a profile created by 2.38.2, those rows were the only record that the keypair was keycard-backed:

- `keypairs.cold_wallet` arrives earlier via `1775586970_add_xpub_to_keypair`, `DEFAULT ''`. Nothing between the two writes it; the only migration in that range is `1779444743_create_preferences`.
- 2.38.2's mobile onboarding never sets it either. It calls `RestoreAccountAndLogin` with a keycard instance UID, and the only writers of a non-empty `ColdWallet` are the explicit migrate action and device sync.

So the drop removes the evidence without moving it.

## Expected

The drop migration backfills `keypairs.cold_wallet` from the `keycards` rows before removing them. A keypair that had keycard rows stays resolvable as cold-wallet-backed afterwards.

## Actual

`cold_wallet` stays `''`. 2.39 reads keycard-ness from that column alone: `isKeycardAccount` returns `kp.coldWalletType == ColdWalletTypeStatusKeycard`, and `KeypairDto.migratedToColdWallet()` is `coldWalletType != ""`. Both are false after the upgrade, so every account of an upgraded profile reads as not a Keycard account.

On device, after upgrading 2.38.2 `3ef1715600` to `2.39.0-rc.2-12-g2105190783` with a real card:

| Where | Shows |
|---|---|
| Wallet account row | Keycard badge gone |
| Settings → Wallet | Profile keypair listed as **"On device"** |
| Keypair menu | Offers **"Move key pair to a Keycard"** on a keypair already on one |

Login and authentication still use the card, so this isn't a lockout. The profile-level flag comes from `keycardPairing` in the multiaccounts DB, which these migrations don't touch.

Profiles created on 2.39 aren't affected. `pkg/backend/geth_backend.go:2139` sets `coldWallet = ColdWalletTypeStatusKeycard` when a keycard instance UID is present.

## Steps to reproduce

On a phone, with a real Keycard:

1. Install 2.38.2. Create a profile on an empty Keycard.
2. Open Wallet. The account row shows a Keycard badge, and Settings → Keycard lists the card.
3. Install 2.39 over the top. Don't uninstall first, or you lose the profile and the bug with it.
4. Log in with the card. This still works.
5. Open Wallet, then Settings → Wallet, then the keypair's menu.

Step 2 is the before, step 5 the after. The table above lists what changes.

Or at the database, with no device or card:

1. `sqlite.OpenDB` in-memory, then `migrationsprevnodecfg.Migrate` + `nodecfg.MigrateNodeConfig`.
2. `migrations.MigrateTo(db, customSteps, 1779444743)`.
3. Seed a pre-refactor keycard profile:
   ```sql
   INSERT INTO keypairs (key_uid, name, type, derived_from)
     VALUES ('0xkeycarduser', 'Keycard keypair', 'profile', '0xkeycarduser');
   INSERT INTO keycards (keycard_uid, keycard_name, keycard_locked, key_uid, position)
     VALUES ('a1b2c3d4', 'My keycard', 0, '0xkeycarduser', 0);
   ```
4. `migrations.Migrate(db, customSteps)`.
5. `SELECT cold_wallet FROM keypairs WHERE key_uid = '0xkeycarduser';`

Every migration passes and the select returns `''`.

## Notes

One more consequence, traced but not seen, so treat it as a lead. With the account flag false and the profile flag still true, the client sets `externalAuthorization` and takes the authorise-then-sign branch instead of sign-on-card. That branch ends at `SignMessage` (`pkg/services/wallet/api.go:570`) calling `LoadAccount`, which wants a local private key the profile doesn't have. Send, swap, community token and dApp signing all share it. We haven't tried a transaction on an upgraded profile yet.

Close this if clients are meant to re-derive keycard state from somewhere else.

### PR DESCRIPTION
**Pull request by saledjenic**
Before dropping the keycard tables, update the `cold_wallet` column for the key pairs that had associated keycards.

### CODE REVIEW

**Review by friofry — APPROVED** [review]

**Review by alexjba — APPROVED** [review]

File: internal/db/appdatabase/database.go [hunk 1]
```diff
@@ -463,3 +465,23 @@ func migrateWalletTransferFromToAddresses(sqlTx *sql.Tx) error {
 
 	return nil
 }
+
+// backfillKeycardColdWalletFromKeycardTables runs right before 1779877215_drop_keycard_tables.
+//
+// Before dropping the keycard tables, update the `cold_wallet` column for the key pairs
+// that had associated keycards befoire 1779877215_drop_keycard_tables migration.
+func backfillKeycardColdWalletFromKeycardTables(sqlTx *sql.Tx) error {
```
Comment (friofry) [inline internal/db/appdatabase/database.go:473]: It's riskier than it looks, please don't go with the `PreMigration`. We should follow canonical migration approach (even it's one-way) to avoid edge-cases. If you wan't I can provide tests and show why it's error prone.

Could you try a plain SQL file instead, numbered between `1779444743` and `1779877215`, with no Go changes:
```sql
-- 1779877214_backfill_keypairs_cold_wallet_from_keycards.up.sql

UPDATE keypairs
SET cold_wallet = 'status-keycard'
WHERE cold_wallet = ''
  AND key_uid IN (SELECT key_uid FROM keycards WHERE keycard_uid IS NOT NULL AND keycard_uid != '');
```

It should do the job. DBs on 2.38 or older run it right before the tables are dropped, and DBs already past `1779877215` skip it, same as the pre-step does now. The file runs in its own transaction, and the shipped `1779877215` stays untouched. I ran your `TestDropKeycardTablesBackfillsColdWallet` with this file and without the `migrate.go` / `database.go` changes: green with the file, red without it.

**Review by friofry — CHANGES_REQUESTED** [review]

File: internal/db/appdatabase/database.go (hunk 1 above)
Comment (saledjenic) [reply internal/db/appdatabase/database.go:473]: That's how I did it at first, and it worked, but it didn't pass the checks.
`check_migration_order()` failed in that case, so I changed my approach.

Comment (saledjenic) [reply internal/db/appdatabase/database.go:473]: Also that's the way we were using in the past, just for migrations that run after the version has been applied, this is the same approcah, but for migrations that need to be applied before the version.

Comment (friofry) [reply internal/db/appdatabase/database.go:473]: I would avoid adding API for this kind of patches in future. It shouldn't be a "convenient" mechanism.

How about making it in a `doMigration` manually?

Here is an example: https://github.com/status-im/status-go/commit/62ee1cff74ef8a949662567d3f3c954a14d73662

**Review by friofry — APPROVED** [review]
LGTM, I would just make it less easy to add this kind of fixes in future

File: internal/db/appdatabase/database.go (hunk 1 above)
Comment (saledjenic) [reply internal/db/appdatabase/database.go:473]: @friofry I see your point, but the way it's already done is aligned with the custom migration we already have.

File: internal/db/sqlite/migrate.go
Comment (friofry) [inline internal/db/sqlite/migrate.go:91]:
```
	for i, step := range options.CustomSteps {
		if step.PreMigration == nil {
			continue
		}
		if step.PreMigrationVersion == 0 || step.PreMigrationVersion >= step.Version ||
			(i > 0 && step.PreMigrationVersion < options.CustomSteps[i-1].Version) {
			return fmt.Errorf("invalid PreMigrationVersion %d for custom step %d", step.PreMigrationVersion, step.Version)
		}
	}
```

against misconfigured steps

Comment (saledjenic) [reply internal/db/sqlite/migrate.go:91]: Updated.

**Comment by saledjenic** [comment]
The first commit was cherry picked https://github.com/status-im/status-go/commit/fbcbc341922e6f9106c79747602c92db984a26af no need for devops approvals here.
